crates/proto/src/runtime_provider.rs

```diff
@@ -0,0 +1,72 @@
+//! Runtime abstraction component for DNS clients
+use crate::{error::ProtoError, tcp::DnsTcpStream, udp::UdpSocket, Time};
+use std::{future::Future, io, net::SocketAddr, pin::Pin};
+
+/// RuntimeProvider defines which async runtime that handles IO and timers.
+#[async_trait::async_trait]
```

Comment:
Ok, the only question left now is if this annotation is desirable. As far as I can tell it's only useful in the `async fn` on bind_udp, and that could be just as easily `Box::pin`ed like the tcp function. Thoughts? Otherwise I think this is all ready.